spyderlib/spyder.py

```diff
@@ -2735,6 +2754,7 @@ def restart(self):
         env['SPYDER_ARGS'] = spyder_args
         env['SPYDER_PID'] = str(pid)
         env['SPYDER_IS_BOOTSTRAP'] = str(is_bootstrap)
+        env['SPYDER_RESET'] = str(reset)

```

Comment:
adding `env['PYTHON_PATH'] = sys.path` gets rid of the import errors, but it seems the old Spyder is not shutting down though.